database/postgres/postgres.go

```diff
@@ -358,6 +358,38 @@ func (p *Postgres) Drop() (err error) {
 			}
 		}
 	}
+	query = `SELECT typname FROM pg_type WHERE typcategory = 'E';`
```

Comment:
Looks like each [pg_type](https://www.postgresql.org/docs/current/catalog-pg-type.html) has its own [pg_namespace](https://www.postgresql.org/docs/current/catalog-pg-namespace.html) e.g. schema
So you'll need to scope this query to the configured namespace/schema.

Reply:
So I think the pg_namespace is for schema. It is possible to the following SQL in the same database.
```
CREATE SCHEMA test_schema;
CREATE TYPE test_schema.test_enum AS ENUM ('value1', 'value2');
CREATE TYPE test_enum AS ENUM ('value3', 'value4');
```

So there is a lot more information that can be dropped besides enums,  and. this merge request does not handle all of it, but this might be one of the more used items.

Reply:
We should only drop enums from the configured namespace/schema.
We need to assume that some users run on a shared DB and can't drop schemas owned by others.

Reply:
I understand your concern but every namespace/schema has it's own pg_type table. We are selecting from the table in the namespace/schema that we want dropped. As opposed to selecting users from the pg_catalog schema see example query below. 
```
SELECT usename FROM pg_catalog.pg_user;
```

As I mentioned above I haven't seen clear documentation that clearly states this other than the fact that we are selecting from the table in the same database, and the tests I have run.

See below
```
postgres=# create type a as enum ('A');
CREATE TYPE
postgres=# SELECT typname FROM pg_type WHERE typcategory = 'E';
 typname
---------
 a
(1 row)

postgres=# create database test;
CREATE DATABASE
postgres=# \c test
psql (13.1, server 11.9 (Debian 11.9-1.pgdg90+1))
You are now connected to database "test" as user "postgres".
test=# SELECT typname FROM pg_type WHERE typcategory = 'E';
 typname
---------
(0 rows)
```

Reply:
Ahhh, thanks for point this out!
The [docs](https://www.postgresql.org/docs/current/catalogs-overview.html) state that:
> Most system catalogs are copied from the template database during database creation and are thereafter database-specific. A few catalogs are physically shared across all databases in a cluster; these are noted in the descriptions of the individual catalogs.

`pg_type` doesn't [have](https://www.postgresql.org/docs/current/catalog-pg-type.html)  this callout and for a sanity check `pg_database` [does](https://www.postgresql.org/docs/current/catalog-pg-database.html):
> Unlike most system catalogs, `pg_database` is shared across all databases of a cluster: there is only one copy of `pg_database` per cluster, not one per database.